core/src/test/java/org/apache/iceberg/rest/TestExponentialHttpRequestRetryStrategy.java

```diff
@@ -207,16 +198,22 @@ public void invalidRetryAfterHeader() {
   }
 
   @Test
-  public void testRetryBadGateway() {
-    HttpContext context = new BasicHttpContext();
-    BasicHttpResponse response502 = new BasicHttpResponse(502, "Bad gateway failure");
-    assertThat(retryStrategy.retryRequest(response502, 3, context)).isTrue();
-  }
+  public void testRetryHappensOnAcceptableStatusCodes() {
```

Comment:
Is it possible to separate into two separate tests, one for acceptable codes and another for status codes we do not want to retry? and then could we parameterize that via `@ParameterizedTest` with a MethodSource of the codes? Then it becomes clear which codes we're testing and their expectations, along with reducing a bit of duplication